### PR TITLE
YahooWeather: switch to the new system channels

### PR DESCRIPTION
**Pull request by lolodomo**
Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
@cweitkamp for review.
Note that there is no description defined for the system channels so we loose the channel description in the binding too.

File: extensions/binding/org.eclipse.smarthome.binding.yahooweather/ESH-INF/i18n/yahooweather_de.properties [hunk 1]
```diff
@@ -9,11 +9,3 @@ thing-type.config.yahooweather.weather.location.label = Ort
 thing-type.config.yahooweather.weather.location.description = Ort der Wetterinformation.
 thing-type.config.yahooweather.weather.refresh.label = Aktualisierungsintervall
 thing-type.config.yahooweather.weather.refresh.description = Spezifiziert das Aktualisierungsintervall in Sekunden
-
-# channel types
-channel-type.yahooweather.temperature.label = Temperatur
-channel-type.yahooweather.temperature.description = Aktuelle Temperatur
```
Comment (cweitkamp) [inline extensions/binding/org.eclipse.smarthome.binding.yahooweather/ESH-INF/i18n/yahooweather_de.properties]:
```java
thing-type.yahooweather.weather.channel.temperature.description = Aktuelle Temperatur.
```

File: extensions/binding/org.eclipse.smarthome.binding.yahooweather/ESH-INF/thing/thing-types.xml [hunk 2]
```diff
@@ -9,9 +9,9 @@
 		<description>Provides various weather data from the Yahoo service</description>
 
 		<channels>
-			<channel id="temperature" typeId="temperature" />
-			<channel id="humidity" typeId="humidity" />
-			<channel id="pressure" typeId="pressure" />
+			<channel id="temperature" typeId="system.outdoor-temperature" />
```
Comment (cweitkamp) [inline extensions/binding/org.eclipse.smarthome.binding.yahooweather/ESH-INF/thing/thing-types.xml]:
```java
<channel id="temperature" typeId="system.outdoor-temperature">
    <description>Current temperature.</description>
</channel>
```

**Comment by lolodomo** [comment]
Are we ok that the system channels temperature, humidity and pressure are for current data ?

**Comment by lolodomo** [comment]
I am going to create a PR to add a description to these system channels.

**Pull request merged**